File: src/etch/bindings/java/transport/BinaryTaggedDataFormatFactory.java
package etch.bindings.java.transport;

import metreos.util.URL;
import etch.bindings.java.msg.TaggedDataInput;
import etch.bindings.java.msg.TaggedDataOutput;
import etch.bindings.java.msg.ValueFactory;
import etch.bindings.java.transport.fmt.binary.BinaryTaggedDataInput;
import etch.bindings.java.transport.fmt.binary.BinaryTaggedDataOutput;

/**
 * Format factory for binary tagged data.
 */
public class BinaryTaggedDataFormatFactory extends FormatFactory
{
	@Override
	public TaggedDataInput newTaggedDataInput( ValueFactory vf, URL uri )
	{
		return new BinaryTaggedDataInput( vf, uri );
	}

	@Override
	public TaggedDataOutput newTaggedDataOutput( ValueFactory vf, URL uri )
	{
		return new BinaryTaggedDataOutput( vf, uri );
	}
}

File: src/etch/bindings/csharp/transport/BinaryTaggedDataFormatFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Transport
{
    public class BinaryTaggedDataFormatFactory : FormatFactory
    {
        public override TaggedDataInput NewTaggedDataInput( Etch.Msg.ValueFactory vf, Etch.Util.URL uri )
        {
            return new Etch.Transport.Fmt.Binary.BinaryTaggedDataInput( vf, uri );
        }

        public override TaggedDataOutput NewTaggedDataOutput( Etch.Msg.ValueFactory vf, Etch.Util.URL uri )
        {
            return new Etch.Transport.Fmt.Binary.BinaryTaggedDataOutput( vf, uri );
        }
    }
}
